emacs/README.md

```diff
@@ -0,0 +1,69 @@
+# zeek-language-server
+
+This directory contains the configuration files needed to tie into `zeek-language-server` from emacs. These instructions use `lsp-mode`. There's the possibility it would also work with `elgot` (an `lsp-mode` replacement), but that exercise is left to the reader. `lsp-mode` comes with an astonishing amount of options for configuring the look and operation of the LSP features in emacs. This guide only presents the most basic configuration.
+
+This guide assumes that you have installed `zeek-language-server` and that it is in your `PATH`.
+
+## Prerequisites
+
+There are some emacs packages that are required for this to work. You can install all of these from your favorite emacs package repository using `M-x package-install`. The required packages are:
+
+```
+use-package
+lsp-mode
+lsp-ui
+yasnippet
+```
+
+It also requires the Zeek major mode from https://github.com/zeek/emacs-zeek-mode. Either copy the file into a directory in your emacs `load-path` or load the file manually like below in your `.emacs`:
+
+```
+(load-file "/path/to/zeek-mode")
+```

```

Comment:
Is this actually a hard requirement, or just nice to have? Since this file isn't in the upstream package repos this seems to introduce some complications.

If we include it we need to make sure that formatting from `emacs-zeek-mode` does not interfere with formatting from the LSP server. Do you know whether this is disabled in `lsp-mode` by default (which would be fine)?

Reply:
It is, because we need zeek-mode for the hook to lsp-mode to work.

Reply:
> As for the formatting, I'm not sure. I'd have to play with it.

Looks like format-on-save is by default also disabled in `emacs-zeek-mode`, so the default config has no conflict. Unless behavior of one of the involved plugins change, there's probably no issue right now unless users go out of their way to mess things up.

> My only concern with doing that is that it's different from how most major modes handle things.

I am also wondering how that would impact users' ability to configure LSP integration. Like you write, this plugin comes with **a lot** of options, and users will likely want to tweak things.